src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1003,6 +1003,154 @@ void Lowering::LowerFusedMultiplyAdd(GenTreeHWIntrinsic* node)
     }
 }
 
+//----------------------------------------------------------------------------------------------
+// TryLowerConstVec: Unfold the Vector to Broadcast->CreateScalarUnsafe->ConstScalar form
+//                   when the constant vector has all the same elements and is an operand
+//                   of an embedded broadcast compatible instruction.
+//
+//  Arguments:
+//     node - The constant vector node
+GenTree* Lowering::TryLowerConstVec(GenTreeVecCon* node)
+{
+    LIR::Use            use;
+    bool                foundUse         = BlockRange().TryGetUse(node, &use);
+    GenTreeHWIntrinsic* embBroadcastNode = nullptr;
+    if (foundUse && use.User()->OperIs(GT_HWINTRINSIC) && use.User()->AsHWIntrinsic()->OperIsEmbBroadcastCompatible())
+    {
+        embBroadcastNode                = use.User()->AsHWIntrinsic();
+        var_types   simdType            = embBroadcastNode->TypeGet();
```

Comment:
`embBroadcastNode` here is a confusing name, IMO.

Its rather the `parentNode` or `user` as we refer to it in a couple other places.